### DISCUSSION TITLE
关于输入法振动问题（算问题吗）

### DISCUSSION THREAD

**Discussion by Entaliga** [fcitx5-android/fcitx5-android#621] [Ideas | 想法]
个人比较喜欢的振动是那种哒哒哒的振动。

但是有一个问题，使用系统默认振动之后，震感是特别大的，而调整的振动（比如3ms）都是嗡嗡嗡。

作为一个小白，想提问，能否将系统默认振动的档位调小，或者说怎么通过修改安装包的形式满足我的需求

**Comment by rocka** [comment]
“系统默认”之所以是默认就是因为它是不能改的。控制振动反馈的代码在这里

https://github.com/fcitx5-android/fcitx5-android/blob/2c0c4c98220c91ea3ae208a5b9a3ac53272171d0/app/src/main/java/org/fcitx/fcitx5/android/data/InputFeedbacks.kt#L56-L94

“系统默认”就是调用 [View#performHapticFeedback](https://developer.android.com/reference/android/view/View#performHapticFeedback(int)) ，而如果修改了振动时间就改为调用 [Vibrator#vibrate](https://developer.android.com/reference/android/os/Vibrator#vibrate(android.os.VibrationEffect)) 。如果你想自己控制振动反馈的波形，可以研究一下这个文档： https://developer.android.com/develop/ui/views/haptics/custom-haptic-effects

**Reply by Entaliga** [reply]
ok，非常感谢了
